plugins/completion/copilot-vim.nix

```diff
@@ -12,33 +12,72 @@ with helpers.vim-plugin;
     originalName = "copilot.vim";
     defaultPackage = pkgs.vimPlugins.copilot-vim;
     globalPrefix = "copilot_";
+
+    # TODO introduced 2024-03-02: remove 2024-05-02
     deprecateExtraConfig = true;
+    optionsRenamedToSettings = [
+      "nodeCommand"
+      "filetypes"
+      "proxy"
+    ];
 
-    options = {
-      nodeCommand = mkDefaultOpt {
-        global = "node_command";
+    settingsOptions = {
```

Comment:
this is missing the `enabled` settings. This settings could be useful in some cases, for example when using extra files in the nixvim configuration for certain filetypes, or to dynamically enable/disable the plugin (plus it's a documented upstream option)

Reply:
The `enabled` setting is a `buffer` option, not `global` ([`b:copilot_enabled`](https://github.com/github/copilot.vim/blob/57a0115908895f465eb3476f03a0aaa7096e8fe1/doc/copilot.txt#L64-L67)). So, according to me, it doesn't make sense to declare this option here.

Reply:
It is especially made to locally override the `filetypes` option.

Reply:
Yes indeed!